pytest_base_url/plugin.py

```diff
@@ -21,13 +24,21 @@ def base_url(request):
 def _verify_url(request, base_url):
     """Verifies the base URL"""
     verify = request.config.option.verify_base_url
+    verify_timeout = request.config.getoption('verify_base_url_attempts')
     if base_url and verify:
-        response = requests.get(base_url, timeout=10)
-        if not response.status_code == requests.codes.ok:
-            raise pytest.UsageError(
-                'Base URL failed verification!'
-                '\nURL: {0}, Response status code: {1.status_code}'
-                '\nResponse headers: {1.headers}'.format(base_url, response))
+        session = requests.Session()
+        retries = Retry(total=verify_timeout,
+                        backoff_factor=0.1,
+                        status_forcelist=[500, 502, 503, 504])
+        session.mount(base_url, HTTPAdapter(max_retries=retries))
+        try:
+            session.get(base_url)
+        except Exception:
```

Comment:
Only if it's possible, or if not maybe the list. I'm not familiar with the potential exceptions. I would say the ideal situation would be:

We detect the exception type when the connection fails (not status code), and we raise an appropriate exception.

We detect the exception type (or can otherwise determine that the exception was due to a failed status code), and raise a suitable exception with the specific status code that was returned. Failing that, include a list of the status code we consider to be failure codes.

If we can't do any of that, then we could just craft a custom exception message that contains everything we do know. It may even be worthwhile including the original message from the raised exception.